### PR TITLE
Add support for StreamArgs

### PR DESCRIPTION
**Pull request by travisperson**
New argument type StreamArgs which will create a file object from stdin.

### CODE REVIEW

File: commands/argument.go [hunk 1]
```diff
@@ -37,6 +38,16 @@ func FileArg(name string, required, variadic bool, description string) Argument
 	}
 }
 
+func StreamArg(name string, required, variadic bool, description string) Argument {
```
Comment (jbenet) [inline commands/argument.go]: maybe comment what `StreamArg` is used for? Im not sure myself atm. 

(i know the entire lib isnt well commented yet... perhaps we should lint the entire codebase)

Comment (travisperson) [inline commands/argument.go]: Yep totally! (btw this is being merged in to feat/pipe not master if you didn't see that, just wanted a easy way to get this to @whyrusleeping).

**Comment by whyrusleeping** [comment]
@jbenet a stream arg will be used when we want to be able to pipe data through our commands. Right now the way we do it is by calling stdin one of our input files. Having inputs labeled as streams gives us a little more freedom in how to work with them.

The main use-case for this right now is the `ipfs pipe` command i'm working on

File: commands/argument.go (hunk 1 above)
Comment (jbenet) [inline commands/argument.go]: > this is being merged in to feat/pipe not master if you didn't see that

ah, missed that. :+1:

**Pull request merged**